cluster-autoscaler/clusterstate/api/utils.go

```diff
@@ -90,3 +99,65 @@ func (status ClusterAutoscalerStatus) GetReadableString() string {
 	}
 	return buffer.String()
 }
+
+// LogItem is a single entry in log managed by LogCollector.
+type LogItem struct {
+	// Log is the logged message body.
+	Log string
+	// Timestamp when the Log was created.
+	Timestamp time.Time
+}
+
+// LogCollector keeps recent log events. It is automatically truncated on each access based on predefined set of conditions.
+type LogCollector struct {
```

Comment:
This is not a part of any api. Lets move the whole file to ./clusterstate/utils/